ecosystem/indexer-grpc/indexer-grpc-table-info/src/internal_indexer_db_service.rs

```diff
@@ -59,8 +59,53 @@ impl InternalIndexerDBService {
         Arc::clone(&self.db_indexer)
     }
 
-    pub async fn run(&mut self) {
-        let mut start_version = self.db_indexer.get_persisted_version().unwrap_or(0);
+    pub fn get_start_version(&self, node_config: &NodeConfig) -> Version {
+        let indexer_version = self
+            .db_indexer
+            .get_persisted_version()
+            .unwrap_or_else(|err| {
+                panic!(
+                    "Failed to get persisted version from internal indexer db: {}",
+                    err
+                )
+            });
+        let fast_sync_enabled = node_config
+            .state_sync
+            .state_sync_driver
+            .bootstrapping_mode
+            .is_fast_sync();
+        let mut db_min_version = self
+            .db_indexer
+            .get_main_db_lowest_viable_version()
+            .unwrap_or_else(|err| panic!("Failed to get main db min viable version: {}", err));
+
+        // Wait till fast sync is done
+        while fast_sync_enabled && db_min_version == 0 {
+            db_min_version = self
+                .db_indexer
+                .get_main_db_lowest_viable_version()
+                .unwrap_or_else(|err| panic!("Failed to get main db min viable version: {}", err));
+            std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
let's not block in an async env. make this method async and call tokio::time::sleep()